### ISSUE TITLE
[BUG] test context is not passed to subtest if accessed in beforeEach

### ISSUE THREAD

**Issue by michele-amati** [tapjs/tapjs#1016]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you read the `CONTRIBUTING` guide on posting bugs, and `CODE_OF_CONDUCT`?

- [X] yes I read the things

### This issue exists in the latest tap version

- [X] I am using the latest tap

### Description

The context I set in my test is not accessible in its subtests IF `beforeEach` is implemented AND I access the test context in it (regardless if I write to it or just read it).

I expected to be able to access the context in the subtest AND read/write it in the `beforeEach`, as it worked in version 16.

### Reproduction

Run the following directly with node (or through tap, doesn't matter).

This fails because t0 context is not passed to t1.
```
const t = require('tap')

t.beforeEach(t => {
  console.log("beforeEach", JSON.stringify(t.context))
})

t.test('lvl0', {}, async t0 => {
  t0.context.aaa = 'aaa'
  t0.test('lvl1', async t1 => {
    t1.equal(t1.context.aaa, 'aaa', 'context is passed on to t1')
  })
})
```

This works, t0 context is passed to t1. It differs from the failing example in that I'm not accessing `t.context` in `beforeEach`.
```
const t = require('tap')

t.beforeEach(t => {
  console.log("beforeEach", "whatever just dont access t.context")
})

t.test('lvl0', {}, async t0 => {
  t0.context.aaa = 'aaa'
  t0.test('lvl1', async t1 => {
    t1.equal(t1.context.aaa, 'aaa', 'context is passed on to t1')
  })
})
```

### Environment

```
tap@18.7.2
Microsoft Windows NT 10.0.22631.0 x64
```

**Comment by isaacs** [comment]
This should be fixed by b4fb6305, in the latest release.

**Comment by michele-amati** [comment]
> This should be fixed by [b4fb630](https://github.com/tapjs/tapjs/commit/b4fb6305a77c4575126fd37e8e73f0237496bd70), in the latest release.

I'm using the latest version available on NPM (18.7.2) and it's not working for me. I've tried forking the repo to add a test demonstrating the issue, but I'm having troubles with the bootstrap script, I'll try again as soon as I have some time. In the meantime, here's a mini demo project showing the problem.

[testap.zip](https://github.com/tapjs/tapjs/files/15151656/testap.zip)
Just npm install and npm run test
